gnomad_constraint/constraint/constraint_pipeline.py

```diff
@@ -0,0 +1,77 @@
+# noqa: D100
+
+import argparse
+import logging
+
+import hail as hl
+
+from gnomad.resources.grch37.gnomad import public_release
+from gnomad_constraint.resources.resource_utils import (
+    preprocessed_ht,
+    get_logging_path,
+    annotated_context_ht,
+)
+from gnomad_constraint.utils.constraint_basics import (
+    add_vep_context_annotations,
+    prepare_ht_for_constraint_calculations,
+)
+
+logging.basicConfig(
+    format="%(asctime)s (%(name)s %(lineno)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
+)
+logger = logging.getLogger("constraint_pipeline")
+logger.setLevel(logging.INFO)
+
+
+def main(args):
+    """Execute the constraint pipeline."""
+    try:
+        if args.add_annotations:
+            logger.info("Adding VEP context annotations...")
+            # Add annotations from VEP context Table to gnomAD data.
+            # TODO: Need to add function that annotates methylation, coverage, and gerp in the vep context table.
+            preprocessed_genome_ht = add_vep_context_annotations(
+                public_release("genomes").ht(), annotated_context_ht.ht()
+            )
+            preprocessed_exome_ht = add_vep_context_annotations(
+                public_release("exomes").ht(), annotated_context_ht.ht()
+            )
+            # Filter input Table and add annotations used in constraint calculations.
+            full_context_ht = prepare_ht_for_constraint_calculations(
+                annotated_context_ht.ht()
+            )
+            full_genome_ht = prepare_ht_for_constraint_calculations(
+                preprocessed_genome_ht
+            )
+            full_exome_ht = prepare_ht_for_constraint_calculations(
+                preprocessed_exome_ht
+            )
+            # Filter to locus that is on an autosome or a pseudoautosomal region in context Table, exome Table, and genome Table
+            full_context_ht.filter(full_context_ht.locus.in_autosome_or_par()).write(
```

Comment:
I moved checkpoint to here for two reasons:
1. context table and exome table are used a lot in the downstream. From my testing experience, it's much more convenient to have exome and context table ready for downstream functions than start from `preprocessed_exome_ht` and `annotated_context_ht `.
2. Technically from beginning to this point is the preprocessing part (adding annotations), so it makes more sense to have a checkpoint here.